### PR TITLE
Task 2 - all elements of learning outcome minus Page Factory

### PR DESCRIPTION
**Pull request by William-436**
Task 2 Learning Outcomes: spreadsheets, POM, data-driven approach, NUnit, Extent Reports, Excel Data Reader, AutoIt, and Asserts

### CODE REVIEW

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/HomePage.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        // page objects
+        private IWebElement managelistingsButton => driver.FindElement(By.XPath(managelistingsButtonXP));
+        private IWebElement shareskillButton => driver.FindElement(By.XPath(shareskillButtonXP));
+
+        // Miscellaneous objects
+        private IWebElement signoutButton => driver.FindElement(By.XPath(signoutButtonXP));
+        private IWebElement signinLink => driver.FindElement(By.XPath(signinLinkXP));
+
+        // find the login button
+        //[FindsBy(How = How.XPath, Using = "/html/body/div[2]/div/div/div[1]/div/div[4]/button")]
```
Comment (amanbansalnz) [inline MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/HomePage.cs]: remember to remove all unwanted comments

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ManageListingsPage.cs [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+        public void NavigateToShareSkill()
+        {
+            // find and click the Share Skill button
+            shareskillButton.Click();
+        }
+
+        // this method is only called by positive test cases containing valid data
+        public bool CompareManageListingsValues(int NumberOfExistingListingsInt, string Category, string Title, string Description, string ServiceType, string SkillTrade, string Active)
+        {
```
Comment (amanbansalnz) [inline MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ManageListingsPage.cs]: can you please break this method into smaller methods, this way your code is more readable and easy to understand

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs [hunk 3]
```diff
@@ -0,0 +1,1144 @@
...
+        //public static string savedMessage;
+        //public static string capturedMessage;
+
+    
+        public void CreateShareSkill(string ActionOfTest, string ValidInvalid, string ActionButton, string FollowupAction, string Title, string TitleError, string Description, string DescriptionError, string Category, string Subcategory, string Tags, string ServiceType, string Location, string StartDate, string EndDate, string StartEndDateError, string Sun, string SunStartTime, string SunEndTime, string Mon, string MonStartTime, string MonEndTime, string Tue, string TueStartTime, string TueEndTime, string Wed, string WedStartTime, string WedEndTime, string Thu, string ThuStartTime, string ThuEndTime, string Fri, string FriStartTime, string FriEndTime, string Sat, string SatStartTime, string SatEndTime, string SkillTrade, string SkillExchange, string Credit, string WorkSamples, string Active, string ExpectedPopupMessage, string ExpectedFollowupPopupMessage)
```
Comment (amanbansalnz) [inline MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs]: so this should be broken down into smaller methods! Also the number of args you are passing in this method is way too much think about other means of doing this! either exposing methods which take less number of arguments or introduce a new object that sets all these fields and pass that object to this method

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs [hunk 4]
```diff
@@ -0,0 +1,1144 @@
...
+            else
+            {
+                Console.WriteLine("FAIL: Invalid time of " + TimetoReformat);
+                test.Log(Status.Fail, "Invalid time of '" + TimetoReformat + "'");
+            }
+            return timeString;
+        }
+
+        public string ConvertDate(string DatetoConvert)
+        {
```
Comment (amanbansalnz) [inline MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs:804]: same with this method, break down into smaller methods also remove comments as its hard to read the code

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        //// for Extent Reports
+        //public static ExtentTest test;
+        //public static ExtentReports extent;
+
+        [SetUp]
+        public static void SetupActions()
+        {
+            Console.WriteLine("**Starting 'Trade Your Skills' portal script from CommonDriver");
+
+            // create the Extent Report
```
Comment (amanbansalnz) [inline MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs]: again break down into readable methods

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+            HomePage homePageObj;
+
+            // initialize Page objects
+            homePageObj = new HomePage();
+
+            if (signoutBool == true) // then attempt to Sign Out
+            {
+                //if (homePageObj.SignOut() == true)
+                //{
+                //    Console.WriteLine("Successfully Signed out of 'Trade Your Skills' portal");
```
Comment (amanbansalnz) [inline MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs]: remove comments

**Comment by William-436** [comment]
The 'Conflicting file' is Task 1's deliverable which I am awaiting feedback on its review. If you approve it without changes, then I am happy to simply remove it from the repository and upload an updated version of the same file containing some corrections and enhancements. Please advise on the status of its review.

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs (hunk 5 above)
Comment (William-436) [reply MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs]: Please clarify reason for breaking down the SetupActions method of the CommonDriver's non-repeatable code into separate methods. I have attempted to explain each portion of code in the SetupActions method with a clear comment. Are the comments unclear or should the portions of code be grouped in a different sequence?

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/HomePage.cs (hunk 1 above)
Comment (William-436) [reply MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/HomePage.cs]: Removed unwanted comments from HomePage.cs

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ManageListingsPage.cs (hunk 2 above)
Comment (William-436) [reply MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ManageListingsPage.cs]: Removed 1 parameter from parms and created separate methods to check each parameter

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs (hunk 3 above)
Comment (William-436) [reply MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs]: Replaced CreateShareSkill method with separated method for each input field

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs (hunk 4 above)
Comment (William-436) [reply MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Pages/ShareSkillPage.cs:804]: Modified ConvertDate method to use a new method (CalculateTheDate) to execute reusable lines of code. Also cleaned up excessive comments and left informative comments.

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs (hunk 6 above)
Comment (William-436) [reply MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs]: Removed unnecessary comments from CommonDriver.cs

**Review by amanbansalnz — APPROVED** [review]

File: MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs (hunk 5 above)
Comment (William-436) [reply MVPSt_2023_Feb_Competition/MVPSt_2023_Feb_Competition/Utilities/CommonDriver.cs]: Resolving conversation as mentor approved submission without requiring this update to be completed.

**Comment by William-436** [comment]
Deleted spreadsheet containing test conditions and test cases from the main branch to resolve merge conflict since newer, updated spreadsheet in the branch is intended to replace the spreadsheet in the main branch.

**Pull request merged**